core/model/objects/asset.hpp

```diff
@@ -34,15 +34,27 @@ class Asset : public AbsObject{
 public:
 
     Asset(
-        const std::string domain,
-        const std::string name,
-        const unsigned long long value,
-        const unsigned int precision
+        const std::string& domain,
+        const std::string& name,
+        const unsigned long long& value,
+        const unsigned int& precision
     );
 
-    json_parse::Object dump();
-    static json_parse::Rule getJsonParseRule();
+    Asset(
+        const std::string& name,
+        const unsigned long long& value
+    );
+
+    using Rule = json_parse::Rule;
+    using Type = json_parse::Type;
+    using Object = json_parse::Object;
+
+    Asset(
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Class 'Asset' has a constructor with 1 argument that is not explicit.](https://www.codacy.com/app/soramitsu/iroha/file/4478128051/issues/source?bid=3712491&fileBranchId=3923596#l52)